CMakeLists.txt

```diff
@@ -436,6 +437,12 @@ if(FLB_COVERAGE)
   set(CMAKE_BUILD_TYPE "Debug")
 endif()
 
+if(FLB_COMPILER_STRICT_POINTER_TYPES)
+  if(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang|AppleClang")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror=incompatible-pointer-types")
```

Comment:
maybe this is a bit cleaner: `add_compile_options(-Wincompatible-pointer-types)` (needs to be tested)

Reply:
I used as follows:

```cmake
if(FLB_COMPILER_STRICT_POINTER_TYPES)
  if(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang|AppleClang")
    add_compile_options(-Werror=incompatible-pointer-types)
  endif()
  # Currently, AppleClang has more struct rules when using -Werror=incompatible-pointer-types.
  # We still permit to discarding const qualifiers warnings.
  if(CMAKE_C_COMPILER_ID MATCHES "AppleClang")
    add_compile_options(-Wno-incompatible-pointer-types-discards-qualifiers)
  endif()
endif()
```

Plus, AppleClang has more struct rules for incompatible pointer types.
Frequently, we discarded for constqualifiers so we still needed to permit this type on incompatibilities on macOS's toolchain.